server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/features/TransportResetFeatureStateAction.java

```diff
@@ -65,4 +83,9 @@ protected void doExecute(Task task, ResetFeatureStateRequest request, ActionList
             feature.getCleanUpFunction().apply(clusterService, client, groupedActionListener);
         }
     }
+
+    @Override
+    protected ClusterBlockException checkBlock(ResetFeatureStateRequest request, ClusterState state) {
+        return null;
```

Comment:
This probably should probably actually check the blocks, could you update this to:

```suggestion
        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
```